### PR TITLE
Decode the address type

### PR DESCRIPTION
**Pull request by Gsantomaggio**
Fixes https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/issues/218

Signed-off-by: Gabriele Santomaggio <G.santomaggio@gmail.com>

### CODE REVIEW

**Comment by Gsantomaggio** [comment]
@bSirotkovic do you have a chance to test it? 

thank you

File: RabbitMQ.Stream.Client/RoutingClient.cs [hunk 1]
```diff
@@ -100,6 +99,25 @@ public static class RoutingHelper<T> where T : IRouting, new()
...
         }
 
+        internal static async Task<EndPoint> GetEndPoint(Broker broker)
+        {
+            switch (Uri.CheckHostName(broker.Host))
+            {
+                case UriHostNameType.Basic:
+                case UriHostNameType.Dns:
+                    var hostEntry = await Dns.GetHostEntryAsync(broker.Host);
+                    var endPointNoLb = new IPEndPoint(hostEntry.AddressList.First(), (int)broker.Port);
```
Comment (lukebakken) [inline RabbitMQ.Stream.Client/RoutingClient.cs:109]: Is it allowed for a user to configure DNS to use the same host name for a cluster, and configure multiple IPs for that host name? If so, we should select a random IP returned here.

**Comment by bSirotkovic** [comment]
@Gsantomaggio 
Will get back to you tomorrow

**Comment by bSirotkovic** [comment]
Hi, the initial connection works great. I pass all of our node endpoints to the StreamSystemConfig and the initial connect is great. Multiple producers connect to different leader nodes.

Haven't had the chance to test out node falling out.
When do you think this will be released?

File: RabbitMQ.Stream.Client/RoutingClient.cs (hunk 1 above)
Comment (Gsantomaggio) [reply RabbitMQ.Stream.Client/RoutingClient.cs:109]: We have `first()` in production for a while https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/RoutingClient.cs#L63

Should we change it?

Comment (lukebakken) [reply RabbitMQ.Stream.Client/RoutingClient.cs:109]: Let's not bother until someone requests the ability to configure a DNS host with multiple IP addresses

**Comment by Gsantomaggio** [comment]
@bSirotkovic, we should release it next week.

**Comment by Gsantomaggio** [comment]
@bSirotkovic Done https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/releases/tag/v1.1.1

**Pull request merged**